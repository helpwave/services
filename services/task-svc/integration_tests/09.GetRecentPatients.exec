#!/bin/bash

echo "First, get a clean slate:"
cat <<EOF | redis-cli > /dev/null
DEL recently-viewed-patient-18159713-5d4e-4ad5-94ad-fbb6bb147984
DEL recently-viewed-ward-18159713-5d4e-4ad5-94ad-fbb6bb147984
EOF

echo "Put some patients in recently viewed:"

echo "Assigned Patient, with tasks:"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"id": "06f57c97-509e-4b41-9839-233f29197f56"}' \
	localhost:3001 \
	proto.services.task_svc.v1.PatientService/GetPatientDetails > /dev/null

sleep 1

echo "Assigned Patient, no tasks:"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"id": "c1c104f0-52c5-46d3-9441-33d0b0c03f87"}' \
	localhost:3001 \
	proto.services.task_svc.v1.PatientService/GetPatientDetails > /dev/null
sleep 1

echo "Unassigned Patient, with tasks:"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"id": "2e830d2f-69c9-4e38-bd4b-ab43decc7ae3"}' \
	localhost:3001 \
	proto.services.task_svc.v1.PatientService/GetPatientDetails > /dev/null

grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{}' \
	localhost:3001 \
	proto.services.task_svc.v1.PatientService/GetRecentPatients
