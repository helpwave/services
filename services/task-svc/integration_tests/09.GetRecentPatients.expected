First, get a clean slate:
Put some patients in recently viewed:
Assigned Patient, with tasks:
Assigned Patient, no tasks:
Unassigned Patient, with tasks:
{
  "recentPatients": [
    {
      "id": "c1c104f0-52c5-46d3-9441-33d0b0c03f87",
      "humanReadableIdentifier": "Iggie Lafflina",
      "room": {
        "id": "b79dc72f-8359-4058-bd7a-e4b7019a5ec3",
        "name": "Sales",
        "wardId": "0f13614c-9043-4af7-a439-cc569ba22639"
      },
      "bed": {
        "id": "cb92bad3-68b2-49c1-bfac-494ff18b57f3",
        "name": "Aquamarine"
      }
    },
    {
      "id": "06f57c97-509e-4b41-9839-233f29197f56",
      "humanReadableIdentifier": "Clerc Levay",
      "room": {
        "id": "80d49a5a-573e-4a7a-a9b4-f8d2431f6f14",
        "name": "Legal",
        "wardId": "51aca99b-2372-40b4-99ed-355dad6c014d"
      },
      "bed": {
        "id": "1d8fbe74-37be-4ec1-beb2-0518b51e1da8",
        "name": "Turquoise"
      }
    },
    {
      "id": "2e830d2f-69c9-4e38-bd4b-ab43decc7ae3",
      "humanReadableIdentifier": "Cloris Dunbavin"
    }
  ]
}
